src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2133,32 +2133,103 @@ void CodeGen::genFMAIntrinsic(GenTreeHWIntrinsic* node)
     // Intrinsics with CopyUpperBits semantics cannot have op1 be contained
     assert(!copiesUpperBits || !op1->isContained());
 
-    if (op2->isContained() || op2->isUsedFromSpillTemp())
+    unsigned overwrittenOpNum = 0;
+    LIR::Use use;
+    if (LIR::AsRange(compiler->compCurBB).TryGetUse(node, &use))
     {
-        // 132 form: op1 = (op1 * op3) + [op2]
+        overwrittenOpNum = node->GetOverwrittenOpNumForFMA(use.User(), op1, op2, op3);
+    }
 
-        ins    = (instruction)(ins - 1);
-        op1Reg = op1->GetRegNum();
-        op2Reg = op3->GetRegNum();
-        op3    = op2;
+    // Intrinsics with CopyUpperBits semantics must have op1 as target
+    if (overwrittenOpNum == 1 || copiesUpperBits)
+    {
+        if (op2->isContained() || op2->IsRegOptional())
+        {
+            // op1 = (op1 * [op2]) + op3
+            // 132 form: XMM1 = (XMM1 * [XMM3]) + XMM2
+            ins    = (instruction)(ins - 1);
+            op1Reg = op1->GetRegNum();
+            op2Reg = op3->GetRegNum();
+            op3    = op2;
+        }
+        else
+        {
+            // op1 = (op1 * op2) + [op3]
+            // 213 form: XMM1 = (XMM2 * XMM1) + [XMM3]
+            op1Reg = op1->GetRegNum();
+            op2Reg = op2->GetRegNum();
+        }
+        isCommutative = copiesUpperBits;
```

Comment:
`isCommutative` should only be present for `213` form (this was even done previously). So this should be inside the previous `else` present for `213 form`. Similarly  below.